pallets/credit/src/lib.rs

```diff
@@ -1033,6 +1136,69 @@ pub mod pallet {
             Self::do_add_credit(account_id.clone(), credit_data);
             Self::deposit_event(Event::StakingCreditScore(account_id, credit));
         }
+
+        fn calc_usdt_daily_poc_reward(
+            account_id: &T::AccountId,
+            credit_data: &CreditData,
+        ) -> (BalanceOf<T>, Weight) {
+            let mut weight = Weight::zero();
+            let staking_balance = Self::user_staking_balance(account_id);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+            if staking_balance.is_none() {
+                return (0u32.into(), weight);
+            }
+            let staking_usdt = staking_balance.unwrap().0;
+            let price = Self::dpr_price();
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+            if price.is_none() {
+                return (0u32.into(), weight);
+            }
+            let dpr_amount =
+                staking_usdt / price.unwrap() * UniqueSaturatedFrom::unique_saturated_from(DPR);
```

Comment:
Whether we should use `staking_usdt * price` here?

if `price` means one 1 USDT equals like `25_000_000_000_000_000` (0.025DPR), staking `x` USDT should got `x * price` DPR.

Reply:
??

Reply:
Sorry, my mistake, you are right.